### PR TITLE
Treat default gate families with tags_to_ignore or tags_to_accept as custom gate families in Gatesets

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/5543

### CODE REVIEW

File: cirq-core/cirq/ops/gateset.py [hunk 1]
```diff
@@ -370,7 +378,7 @@ def __init__(
                 )
 
         for g in unique_gate_list:
-            if type(g) == GateFamily:
+            if type(g) == GateFamily and not (g.tags_to_ignore or g.tags_to_accept):
```
Comment (pavoljuhas) [inline cirq-core/cirq/ops/gateset.py:381]: Please use `isinstance(GateFamily, g)`.

File: cirq-core/cirq/ops/gateset_test.py [hunk 2]
```diff
@@ -416,3 +416,12 @@ def test_gateset_eq():
             )
         )
     )
+
+
+def test_gateset_contains_with_tags_to_ignore():
+    gs = cirq.Gateset(
+        cirq.GateFamily(cirq.ZPowGate, tags_to_accept=["PhysicalZTag"]),
```
Comment (pavoljuhas) [inline cirq-core/cirq/ops/gateset_test.py:423]: Can we add a separate test for the `tags_to_accept` behavior only?

The following is also failing with assertion error in master:
```python
import cirq
import cirq_google as cg

gs = cirq.Gateset(
    cirq.GateFamily(cirq.ZPowGate, tags_to_accept=[cg.PhysicalZTag()]),
)
op = cirq.Z(cirq.LineQubit(1))
print("op in gs =", op in gs)
```

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
Perhaps we should test the behavior of tags_to_ignore/tags_to_accept separately.

**Review by verult — COMMENTED** [review]
Thanks for the fix!

If you don't mind, could you also help me uncomment this test case in this PR? Let me know if that ends up breaking tests. https://github.com/quantumlib/Cirq/blob/70487fdffe2f5190419ea439d29f2831faf72f33/cirq-google/cirq_google/devices/known_devices_test.py#L509-L512

File: cirq-core/cirq/ops/gateset_test.py (hunk 2 above)
Comment (verult) [reply cirq-core/cirq/ops/gateset_test.py:423]: A gateset with only tags_to_ignore is similarly failing on master.

**Comment by tanujkhattar** [comment]
@verult @pavoljuhas I've addressed all comments, this is ready for a re-review.

**Review by verult — APPROVED** [review]
Thanks! LGTM

**Review by pavoljuhas — APPROVED** [review]
LGTM, thank you!

**Pull request merged**